crates/cli/src/sarif/sarif_utils.rs

```diff
@@ -694,6 +694,53 @@ pub fn generate_sarif_report(
         .build()?)
 }
 
+pub fn generate_sarif_file(
+    configuration: &CliConfiguration,
+    static_analysis_rule_results: &[RuleResult],
+    secrets_rule_results: &[SecretResult],
+    sarif_report_metadata: SarifReportMetadata,
+) -> Result<String> {
+    let static_rules_sarif: Vec<SarifRule> = configuration
+        .rules
+        .iter()
+        .cloned()
+        .map(|r| r.into())
+        .collect();
+
+    let secrets_rules_sarif: Vec<SarifRule> = configuration
+        .secrets_rules
+        .clone()
+        .into_iter()
+        .map(|r| r.into())
+        .collect();
+
+    let static_analysis_results = static_analysis_rule_results
+        .iter()
+        .cloned()
+        .map(SarifRuleResult::try_from)
+        .collect::<Result<Vec<_>, _>>()
+        .map_err(anyhow::Error::msg)?;
+
+    let secret_results = secrets_rule_results
+        .iter()
+        .cloned()
+        .map(SarifRuleResult::try_from)
+        .collect::<Result<Vec<_>, _>>()
+        .map_err(anyhow::Error::msg)?;
```

Comment:
I'm not really fond of all these `cloned` invocations, especially since this is at the end of both callsites typically, so it doesn't make much sense to not consume it here imo. You will have to change the call sites to consume these vectors instead. 

```suggestion
pub fn generate_sarif_file(
    configuration: &CliConfiguration,
    static_analysis_rule_results: Vec<RuleResult>,
    secrets_rule_results: Vec<SecretResult>,
    sarif_report_metadata: SarifReportMetadata,
) -> Result<String> {
    let static_rules_sarif: Vec<SarifRule> = configuration
        .rules
        .iter()
        .cloned()
        .map(|r| r.into())
        .collect();

    let secrets_rules_sarif: Vec<SarifRule> = configuration
        .secrets_rules
        .clone()
        .into_iter()
        .map(|r| r.into())
        .collect();

    let static_analysis_results = static_analysis_rule_results
        .into_iter()
        .map(SarifRuleResult::try_from)
        .collect::<Result<Vec<_>, _>>()
        .map_err(anyhow::Error::msg)?;

    let secret_results = secrets_rule_results
        .into_iter()
        .map(SarifRuleResult::try_from)
        .collect::<Result<Vec<_>, _>>()
        .map_err(anyhow::Error::msg)?;
```